source/esbuild-plugin.civet

```diff
@@ -13,16 +15,91 @@ esbuild.build({
   ]
 }).catch(() => process.exit(1))
 ```
+
+Chain civet output into another esbuild plugin, solid for example
+
+@example
+```javascript
+import esbuild from 'esbuild'
+const { solidPlugin } = require('esbuild-plugin-solid');
+import civetPlugin from '@danielx/civet/esbuild-plugin'
+
+esbuild.build({
+  ...,
+  plugins: [
+    civetPlugin(
+      next: solidPlugin()
+    )
+  ]
+}).catch(() => process.exit(1))
+```
 */
 
-{ readFile } := require 'fs/promises'
+{ readFile, writeFile } := require 'fs/promises'
 path := require 'path'
 
 // NOTE: this references the built version of the module, not the source
 { compile } := require "../dist/main.js"
 
-module.exports = {
-  name: 'civet'
+interface Options
+  filter?: RegExp
+  inlineMap?: boolean
+  js?: boolean
+  next?: unknown
+
+// NOTE: this function is named civet so esbuild gets "civet" as the name of the plugin
+civet := (options: Options) ->
+  {
+    filter=/\.civet$/
+    inlineMap=true
+    js=true
+    next
+  } := options
+
+  let nextTransform
+  let tmpPath: null | string
+
+  if next
+    { tmpdir } := require 'os'
+    tmpPath = path.join tmpdir(), 'civet'
```

Comment:
Don't we need to `mkdir` this directory?

I also worry that this will lead to permission issues when multiple people on the same machine try to use this same directory.  Perhaps naming it `` `civet-${process.pid}` `` would be better?